### ISSUE TITLE
Training BatchNorm and Only BatchNorm: On the Expressive Power of Random Features in CNNs

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#82]
*ICLR 2021*

## Motivation

- How expressive are BN layers?
- What role does BN affine parameters play in deep learning optimization and generalization?

## Key Idea

- Freeze all parameters except for BN layers, and study image recognition performance on multiple scale convolutional networks.
- Analyze BN parameter values to inspect their distribution and infer their role in DL optimization / generalization.

## Previous Approach
- BatchNorm
- Training only BatchNorm
- Randomly Initialized Networks (Lottery Ticket Hypothesis)

## Experiments & Analysis

![image](https://user-images.githubusercontent.com/46308407/127645769-8fed31f2-634f-427b-9d15-ecc11430f1f3.png)

- Use ResNet for flexibility scaling. (Depth and Width: #channels)

![image](https://user-images.githubusercontent.com/46308407/127645704-b26cb3a6-1dcf-4793-a2ff-d3cce87d493f.png)

- Freeze BN parameters from multi-scale ResNet models

![image](https://user-images.githubusercontent.com/46308407/127645890-b076c59c-f4d7-45c0-bdd8-31c3988f4af6.png)

- Freeze different part of network parameters and study test set accuracy. 
- Note that BN parameters are more important than **any two** parameters.
- When sufficiently scaled, BN layers are sufficient to achieve relatively high accuracy.

![image](https://user-images.githubusercontent.com/46308407/127646075-3eaf0d3b-50c8-4d1a-b832-794aa31e8276.png)
- Scales up better in depth than width.

![image](https://user-images.githubusercontent.com/46308407/127646121-26868563-60d1-40c0-8e14-640cb4a5fc70.png)
- When only train BN layers, much of the parameters are really close to **zero.**
- This indicates that BN layers implicitly performs **channel pruning.**
- "Important part of network representation is the set of random features it learns to *ignore*."
- Since only BN params are trainable, the network tries to find the best it can get from random initialization, leading to pruned network. (c.f. Lottery Ticket Hypothesis)
- This does not hold when all parameters are trainable. (The network can *actively* learn representations.)

![image](https://user-images.githubusercontent.com/46308407/127646543-8df3bbc0-7864-4543-826f-9f1d9ee9c281.png)
- Although large fraction of scale parameters of BN layers are close to zero, setting them exactly to zero harms accuracy significantly.

![image](https://user-images.githubusercontent.com/46308407/127646639-1bb59d21-dfeb-495f-bb52-b2c1cb649ff0.png)
- Again, when only train BN parameters, a lot of activations are *turned off*.

## Comments
- Just studied CVPR 2021 Normalization tutorial. 
- It seems quite natural that when network depth/width is *(infinitely)* scaled, even when only BN parameters are trainable, the network can perform classification tasks. (Especially easy ones like CIFAR-10)
- The interesting part is about the magnitude of BN parameters. Much of them are close to zero, which means that the model learns to *ignore* some random features support Lottery Ticket Hypothesis and its variants, in that random initialization can have sufficient flexibility/expressiveness and thus choosing the *right* initialization nodes is important.
- Note that Lottery Ticket Hypothesis stated that large networks have upper hand in performance because they have better chance of having *the right* initialization, easy to optimize.
- A key difference though is that LTH showed that random weights themselves are not powerful enough, but rather some are more easy to optimize than others. In this case, optimization on BN parameters play a similar role, in my opinion. (A small movement that tunes the network in *just the right direction*.)
